### PR TITLE
Add application pre-launches to boot loader

### PR DESCRIPTION
**Pull request by alicerunsonfedora**
This PR makes the following changes:

- Adds `applicationShouldLaunchAtLogin()` and `applicationWillLaunchAtLogin` methods to `ASAppRepresentative`
- Adds `gatherAllApplications()` method to `ASBootloaderCoreService`
- Changes `boot` method in the following ways:
  - Adds new `bootView` optional parameter to set the custom boot loader view
  - Checks all applications in AppKit and runs their `applicationWillLaunchAtLogin` if they are permitted to do so.
  - Shows boot screen after events rather than before all events
- Updates documentation and fixes variable name error in Setup Assistant (was `persistent.AS_SETUP_COMPLETE` when it should have been `persistent.AS_COMPLETED_SETUP`)

### CODE REVIEW

File: game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy [hunk 1]
```diff
@@ -17,11 +17,30 @@ init 5 python:
...
+                    apps.append(obj)
+            return apps
         
-        def boot(self, timeout=5, expressSetup=True, disclaimer=None):
-            renpy.call_screen("ASBootloaderView", timeout=timeout)
+        def boot(self, timeout=5, expressSetup=True, disclaimer=None, bootView="ASBootloaderView"):
             if not persistent.AS_COMPLETED_SETUP:
                 ASSetup.startSetup(express=expressSetup, disclaimer=disclaimer)
+
+            for app in self.gatherAllApplications():
```
Comment (sr229) [inline game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy]: might be better to thread startup applications so it doesn't slow down the whole process.

Comment (alicerunsonfedora) [reply game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy]: So probably have a thread for each `applicationWillLaunchAtLogin`?

Comment (sr229) [reply game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy]: no, run a seperate init thread per startup app, this is considerably safer.

File: game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy [hunk 2]
```diff
@@ -26,19 +28,26 @@ init 5 python:
...
+                if AS_REQUIRES_SYSTEM_EVENTS in app.requires and not app.applicationShouldLaunchAtLogin():
+                    print("WARN: %s cannot run its login service because it doesn't have permission to do so." % (app.bundleName,))
+                else:
+                    print("INFO: Skipping %s (%s) login service because it doesn't have one." % (app.bundleName, app.bundleId, ) )
         
         def boot(self, timeout=5, expressSetup=True, disclaimer=None, bootView="ASBootloaderView"):
             if not persistent.AS_COMPLETED_SETUP:
                 ASSetup.startSetup(express=expressSetup, disclaimer=disclaimer)
 
+            print("INFO: Threading any startup services...")
```
Comment (sr229) [inline game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy]: No need for this.

**Comment by alicerunsonfedora** [comment]
To prevent conflicts, it's probably best to merge #7 first and pull those changes in.

**Pull request merged**